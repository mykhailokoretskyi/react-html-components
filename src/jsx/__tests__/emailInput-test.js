jest.mock('react/lib/ReactDefaultInjection');
import React from 'react';
import EmailInput from '../emailInput.jsx';
import renderer from 'react/lib/ReactTestRenderer';

const emailInputWithListeners = (
    <EmailInput required={true}
               label="test label"
               name="test_input"
               id="test"
               changeCallback={function(){}}
               mouseEnterCallback={function(){}}
               mouseLeaveCallback={function(){}}/>
);

const emailInputWithoutListeners = (
    <EmailInput required={true}
               label="test label"
               name="test_input"
               id="test"/>
);

const emailInputWithValue = (
    <EmailInput label="test label"
               name="test_input"
               id="test"
                validate={true}
                errorMessage="wrong"
                successMessage="correct"
               value="value"
               type="text"/>
);

describe('Input', () => {
    it('with listeners matches snapshot', () => {
        const component = renderer.create(emailInputWithListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('without listeners matches snapshot', () => {
        const component = renderer.create(emailInputWithoutListeners);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('with value matches snapshot', () => {
        const component = renderer.create(emailInputWithValue);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
