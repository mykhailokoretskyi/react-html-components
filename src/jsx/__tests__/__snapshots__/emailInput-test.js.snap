exports[`Input with listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onBlur={[Function bound _onBlur]}
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onFocus={[Function bound _onFocus]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="email"
    value="" />
  <label
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;

exports[`Input with value matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className="validate "
    disabled={false}
    id="test"
    name="test_input"
    onBlur={[Function bound _onBlur]}
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onFocus={[Function bound _onFocus]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={false}
    type="email"
    value="value" />
  <label
    className="active"
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error="wrong"
    data-success="correct"
    htmlFor="test" />
</div>
`;

exports[`Input without listeners matches snapshot 1`] = `
<div
  className="input-field col s12">
  <input
    checked={false}
    className=""
    disabled={false}
    id="test"
    name="test_input"
    onBlur={[Function bound _onBlur]}
    onChange={[Function bound _onChange]}
    onClick={[Function bound _onChange]}
    onFocus={[Function bound _onFocus]}
    onMouseEnter={[Function bound _onMouseEnter]}
    onMouseLeave={[Function bound _onMouseLeave]}
    placeholder=""
    required={true}
    type="email"
    value="" />
  <label
    className=""
    dangerouslySetInnerHTML={
      Object {
        "__html": "test label"
      }
    }
    data-error=""
    data-success=""
    htmlFor="test" />
</div>
`;
